### PR TITLE
Add or modify IMAGEPAIRS related files

### PR DESCRIPTION
**Pull request by JaehaKim97**
- Add new dataset type "IMAGEPAIRS".

- We've tested these codes with 128 x 128 input images with SR scale factor 2 and verified that the result of PSNR is the same as conventional PSNR measuring code written by python.

- However, the current code does not support png type image which is commonly used in the SR field. We will continuously update the code.

### CODE REVIEW

File: flutter/pubspec.lock [hunk 1]
```diff
@@ -28,7 +28,7 @@ packages:
       name: archive
       url: "https://pub.dartlang.org"
     source: hosted
-    version: "3.1.6"
+    version: "3.1.2"
```
Comment (d-uzlov) [inline flutter/pubspec.lock:31]: Please, revert changes in `pubspec.lock`.

**Review by d-uzlov — CHANGES_REQUESTED** [review]

File: flutter/pubspec.lock (hunk 1 above)
Comment (JaehaKim97) [reply flutter/pubspec.lock]: We have reverted changes in `pubspec.lock`.

**Comment by JaehaKim97** [comment]
Hi @d-uzlov,

I have tried to match the format of `*.cc` and `*.h` files via `clang-format`, but it continuously shows a mismatch.

It seems that 'line ending characters' occur errors, but we can not figure out what exactly causes the problem.

Could you please inform me what the problem is?

**Comment by d-uzlov** [comment]
@JaehaKim97 you probably use different settings for clang-format.
You can run `make format/clang` to fix it.

**Comment by d-uzlov** [comment]
Issue with line endings can be fixed using `make format/line-endings`.
But it seems right now there are no issues with line endings.

**Review by d-uzlov — APPROVED** [review]

**Pull request closed without merging**